### PR TITLE
AMK01 - AMK08

### PR DESCRIPTION
**Pull request by zach2good**
AMK is my baby and I want to have it finished in a month or so :D

**TODO**
- Find my old 2019 AMK PR: (https://github.com/DarkstarProject/darkstar/pull/6053/files)
- Cardian KI drops
- Cardian fight
- Drag my old confrontation helper (https://github.com/DarkstarProject/darkstar/pull/6198/files)
- Digging minigame
- Testing
- CharVars -> MissionStatus
- Digging text IDs and more digging spots
- Single quote to double quote, everywhere

**Review Guide**
- Confrontation helper
- Small additions to helm and chocobo digging
- Styling, comments and headers in mission files
- Helpers file, holding minigame content
- Fix helpers loading/reloading logic
- Modify getWorldAngle binding to accept xyz

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: scripts/missions/amk/04_Welcome_To_My_Decrepit_Domicile.lua [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [10178] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.STURDY_METAL_STRIP)
+                    mission:setVar(player, 'Prog', 1)
+                end,
```
Comment (claywar) [inline scripts/missions/amk/04_Welcome_To_My_Decrepit_Domicile.lua:50]: As an alternative to mission charvars, could be using get/setMissionStatus

File: scripts/missions/amk/06_An_Errand_The_Professors_Price.lua [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            },
+        },
+
+        [xi.zone.OUTER_HORUTOTO_RUINS] =
+        {
+            ['qm1'] =
+            {
+                -- TODO: Reminder about the orbs
+                onTrigger = function(player, npc)
+                    return mission:messageSpecial(ID.text.CANNOT_ENTER_BATTLEFIELD, xi.ki.RIPE_STARFRUIT)
```
Comment (claywar) [inline scripts/missions/amk/06_An_Errand_The_Professors_Price.lua]: If this is a battlefield qm, this message might not be overriding, and allow for moving back and forth between BCNM entry and that message (messageText, and I think messageSpecial are both much lower than say, event priority)

File: scripts/missions/amk/08_Lender_Beware_Read_the_Fine_Print.lua [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(19, 176)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [19] = function(player, csid, option, npc)
+                    if option == 1 then
+                        print("progress")
```
Comment (claywar) [inline scripts/missions/amk/08_Lender_Beware_Read_the_Fine_Print.lua]: Progress!

File: scripts/missions/amk/06_An_Errand_The_Professors_Price.lua (hunk 2 above)
Comment (zach2good) [reply scripts/missions/amk/06_An_Errand_The_Professors_Price.lua]: In that case, what should I do here, play a regular messageSpecial and return something else?

Comment (claywar) [reply scripts/missions/amk/06_An_Errand_The_Professors_Price.lua]: If it needs to be completely blocking for people on that mission, I'd suggest manually setting the priority.  Something like:
`return mission:messageSpecial(ID.text.CANNOT_ENTER_BATTLEFIELD, xi.ki.RIPE_STARFRUIT):setPriority(1000)`

Here's some of the standard values, so event would take precedence:
```
Action.Priority = {
    Ignore = 1,
    Default = 10,
    ReplaceDefault = 20,
    Message = 50,
    Event = 100,
    Progress = 1000,
}
```

Comment (zach2good) [reply scripts/missions/amk/06_An_Errand_The_Professors_Price.lua]: Didn't know I could do this, thanks!

**Review by claywar — APPROVED** [review]

**Comment by zach2good** [comment]
I think that'll do for now, I'll end up doing more testing and more playthroughs in the next chunk too

**Pull request merged**